insights/parsers/ls.py

```diff
@@ -399,3 +403,37 @@ class LSlaZ(FileListing):
     """
 
     pass
+
+
+@parser(Specs.ls_files)
+class LSlnHFiles(CommandParser, dict):
+    """
+    Parses file information of ``ls -lnH`` command.
+
+    .. note::
+
+        To parse a specific file, its full path should be added to the
+        `ls_lnH_files` spec via `add_filter`.
+        Only paths point to files are acceptable.
+
+    Examples:
+        >>> from insights.core.filters import add_filter
+        >>> from insights.specs import Specs
+        >>> add_filter(Specs.ls_lnH_files, ['/etc/redhat-release', '/var/log/messages'])
+        >>> type(ls_files)
+        <class 'insights.parsers.ls.LSlnHFiles'>
+        >>> "/etc/redhat-release" in ls_files
+        True
+        >>> ls_files["/etc/redhat-release"].all_zero()
+        False
+    """
+
+    def parse_content(self, content):
+        for line in content:
+            try:
+                line = line.strip()
+                self[line.rsplit(None, 1)[-1]] = FilePermissions(line)
```

Comment:
Looks like we are mean to parse for `FilePermissions` info of the given files here, and unlike the other ls parsers to inherit from `FileListing` directly. Will it make more sense to name this parser with `permission` related wording, say `LSlnHFilesPermissions` ?

Reply:
Good question. However, I don't think so, the reason didn't inherit it from the `FileListing` is that, 1. `FileListing` only works for listing files under specified directories; 2. In most cases, listing a specific file, it's not the only purpose to check its permissions, 'size', 'owner`, `date` would be also candidate checkpoints. I prefer to leave it as `LSlnHFiles`.

Reply:
I see, so this parser name `LSlHFiles` ends with `Files` which is diff from other dir listing ones. Then I am good with it.